thrift0.13/src/main/java/com/linecorp/armeria/common/thrift/ThriftProtocolFactories.java

```diff
@@ -77,59 +83,53 @@ public String toString() {
      */
     public static final TProtocolFactory TEXT_NAMED_ENUM = TTextProtocolFactory.get(true);
 
+    /**
+     * A way to lookup the related {@link TProtocolFactory} from a {@link SerializationFormat}.
+     * Users can add new entries at runtime using
+     * {@link #registerThriftProtocolFactory(SerializationFormat, TProtocolFactory)}.
+     */
+    private static final ConcurrentMap<SerializationFormat, TProtocolFactory> knownProtocolFactories =
+            new ConcurrentHashMap<>();
+
+    /**
+     * Registers a new Thrift protocol. This operation cannot be undone.
+     *
+     * @param serializationFormat the handle for this new protocol
+     * @param protocolFactory a factory to instantiate this protocol
+     */
+    public static void registerThriftProtocolFactory(
+            SerializationFormat serializationFormat, TProtocolFactory protocolFactory) {
+        knownProtocolFactories.put(serializationFormat, protocolFactory);
+    }
```

Comment:
How about using SPI instead of this, like others?

Reply:
e.g.

```java
public interface ThriftProtocolFactoryProvider {
    ...
}

public interface ThriftProtocolFactory {
    SerializationFormat serializationFormat();
    TProtocolFactory protocolFactory(); // Or delegate all methods internally, because protocolFactory.protocolFactory() doesn't look nice.
}
```